server/src/main/java/keywhiz/service/daos/SecretContentDAO.java

```diff
@@ -61,7 +61,7 @@
   private final RowHmacGenerator rowHmacGenerator;
   private final KeywhizConfig config;
 
-  private SecretContentDAO(DSLContext dslContext, ObjectMapper mapper,
+  SecretContentDAO(DSLContext dslContext, ObjectMapper mapper,
       SecretContentMapper secretContentMapper, RowHmacGenerator rowHmacGenerator,
       KeywhizConfig config) {
     this.dslContext = dslContext;
```

Comment:
For ease of testing, this constructor is now package private. This avoids use of the associated DAO factory and its built-in behavior.

Reply:
I'll add @VisibleForTesting here as well.